tests/php/Tasks/LinkableMigrationTaskTest.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+namespace SilverStripe\LinkField\Tests\Tasks;
+
+use LogicException;
+use ReflectionMethod;
+use ReflectionProperty;
+use RuntimeException;
+use SilverStripe\Dev\Deprecation;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\LinkField\Models\EmailLink;
+use SilverStripe\LinkField\Models\ExternalLink;
+use SilverStripe\LinkField\Models\FileLink;
+use SilverStripe\LinkField\Models\Link;
+use SilverStripe\LinkField\Models\PhoneLink;
+use SilverStripe\LinkField\Models\SiteTreeLink;
+use SilverStripe\LinkField\Tasks\LinkableMigrationTask;
+use SilverStripe\LinkField\Tests\Tasks\LinkableMigrationTaskTest\HasManyLinkableLinkOwner;
+use SilverStripe\LinkField\Tests\Tasks\LinkableMigrationTaskTest\HasOneLinkableLinkOwner;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\Queries\SQLSelect;
+use SilverStripe\ORM\Queries\SQLUpdate;
+use SilverStripe\Versioned\Versioned;
+
+class LinkableMigrationTaskTest extends SapphireTest
+{
+    private const OLD_LINK_TABLE = 'LinkableMigrationTaskTest_OldLinkTable';
+
+    private const TYPE_MAP = [
+        'URL' => ExternalLink::class,
+        'Email' => EmailLink::class,
+        'Phone' => PhoneLink::class,
+        'File' => FileLink::class,
+        'SiteTree' => SiteTreeLink::class,
+    ];
+
+    protected static $fixture_file = 'LinkableMigrationTaskTest.yml';
+
+    protected static $extra_dataobjects = [
+        HasManyLinkableLinkOwner::class,
+        HasOneLinkableLinkOwner::class,
+    ];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        LinkableMigrationTask::config()->merge('base_link_columns', [
+            'MySort' => 'Sort',
+        ]);
+    }
+
+    public function onBeforeLoadFixtures(): void
+    {
+        LinkableMigrationTask::config()->set('old_link_table', self::OLD_LINK_TABLE);
+        // Set up migration tables
+        DB::get_schema()->schemaUpdate(function () {
+            // Old link table
+            $linkDbColumns = [
+                ...DataObject::config()->uninherited('fixed_fields'),
+                // Fields directly from the Link class
+                'Title' => 'Varchar',
+                'Type' => 'Varchar(50)',
+                'URL' => 'Text',
+                'Email' => 'Varchar',
+                'Phone' => 'Varchar(30)',
```

Comment:
```suggestion
                'Title' => 'Varchar(255)',
                'Type' => 'Varchar',
                'URL' => 'Varchar(255)',
                'Email' => 'Varchar(255)',
                'Phone' => 'Varchar(255)',
```
According to https://github.com/sheadawson/silverstripe-linkable/blob/master/src/Models/Link.php

Reply:
Updated